partner_noncommercial/views/res_partner.xml

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+
+    <record id="view_res_partner_filter" model="ir.ui.view">
+        <field name="inherit_id" ref="base.view_res_partner_filter" />
+        <field name="model">res.partner</field>
+        <field name="mode">extension</field>
+        <field type="xml" name="arch">
+
+            <!-- Default type_person also contains invoice and delivery
+                 addresses for companies (and persons). -->
+            <filter name="type_person" position="attributes">
+                <attribute
+                    name="domain"
+                >[('is_company', '=', False), ('type', '=', 'contact')]</attribute>
+            </filter>
+
+            <!-- Change string of Company filter to Organisations. -->
+            <filter name="type_company" position="attributes">
+                <attribute name="string">Organisations</attribute>
+            </filter>
+
+            <!-- Add extra filters. -->
```

Comment:
Odoo also has a filter where you can type a company name and then search for it:

```<field name="parent_id" domain="[('is_company', '=', True)]" operator="child_of"/>```

Maybe we should also rename that one to Organisation.